SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+  //
+  // auto theEcalEBPhase2TPGPedestals_handle = setup.getHandle(theEcalEBPhase2TPGPedestals_Token_);
+  //const EcalEBPhase2TPGPedestalsMap* ebTPPedestals = theEcalEBPhase2TPGPedestals_handle.product();
+  //
+  edm::ESHandle<EcalTPGCrystalStatus> theEcalTPGCrystalStatus_handle = setup.getHandle(theEcalTPGCrystalStatus_Token_);
+  const EcalTPGCrystalStatus* ecaltpgBadX = theEcalTPGCrystalStatus_handle.product();
+  //
+  edm::ESHandle<EcalEBPhase2TPGAmplWeightIdMap> theEcalEBTPGAmplWeightIdMap_handle =
+      setup.getHandle(theEcalEBTPGAmplWeightIdMap_Token_);
+  const EcalEBPhase2TPGAmplWeightIdMap* ecaltpgAmplWeightMap = theEcalEBTPGAmplWeightIdMap_handle.product();
+  //
+  edm::ESHandle<EcalEBPhase2TPGTimeWeightIdMap> theEcalEBTPGTimeWeightIdMap_handle =
+      setup.getHandle(theEcalEBTPGTimeWeightIdMap_Token_);
+  const EcalEBPhase2TPGTimeWeightIdMap* ecaltpgTimeWeightMap = theEcalEBTPGTimeWeightIdMap_handle.product();
+  //
+  edm::ESHandle<EcalTPGWeightGroup> theEcalTPGWeightGroup_handle = setup.getHandle(theEcalTPGWeightGroup_Token_);
+  const EcalTPGWeightGroup* ecaltpgWeightGroup = theEcalTPGWeightGroup_handle.product();
+  //
+  //edm::ESHandle<EcalTPGTowerStatus> theEcalTPGTowerStatus_handle = setup.getHandle(theEcalTPGTowerStatus_Token_);
+  //const EcalTPGTowerStatus* ecaltpgBadTT = theEcalTPGTowerStatus_handle.product();
+  //
+  //edm::ESHandle<EcalTPGSpike> theEcalTPGSpike_handle = setup.getHandle(theEcalTPGSpike_Token_);
+  //const EcalTPGSpike* ecaltpgSpike = theEcalTPGSpike_handle.product();
+
+  ////////////////
+  algo_->setPointers(ecaltpPed, ecaltpLin, ecaltpgBadX, ecaltpgAmplWeightMap, ecaltpgTimeWeightMap, ecaltpgWeightGroup);
+
+  return setup.get<EcalEBPhase2TPGLinearizationConstRcd>().cacheIdentifier();
+  //  return setup.get<EcalLiteDTUPedestalsRcd>().cacheIdentifier();
+}
+
+void EcalEBTrigPrimPhase2Producer::endRun(edm::Run const& run, edm::EventSetup const& setup) { algo_.reset(); }
+
+EcalEBTrigPrimPhase2Producer::~EcalEBTrigPrimPhase2Producer() {}
+
+// ------------ method called to produce the data  ------------
+void EcalEBTrigPrimPhase2Producer::produce(edm::Event& e, const edm::EventSetup& iSetup) {
+  nEvent_++;
+
+  // get input collections
+  edm::Handle<EBDigiCollectionPh2> barrelDigiHandle;
+
+  if (!e.getByToken(tokenEBdigi_, barrelDigiHandle)) {
+    edm::EDConsumerBase::Labels labels;
+    labelsForToken(tokenEBdigi_, labels);
+    edm::LogWarning("EcalTPG") << " Couldnt find Barrel digis " << labels.module << " and label "
+                               << labels.productInstance << "!!!";
+  }
+
+  if (debug_)
+    std::cout << "EcalTPG"
+              << " =================> Treating event  " << nEvent_ << ", Number of EB digis "
+              << barrelDigiHandle.product()->size() << std::endl;
+
+  auto pOut = std::make_unique<EcalEBPhase2TrigPrimDigiCollection>();
+  auto pOutTcp = std::make_unique<EcalEBPhase2TrigPrimDigiCollection>();
```

Comment:
Is `pOutTcp` not used? Can it be removed?

Reply:
done